src/lib/containers/UserOrTeamBadge.tsx

```diff
@@ -0,0 +1,75 @@
+import { Skeleton } from '@material-ui/lab'
+import React, { useEffect, useState } from 'react'
+import { SynapseClient } from '../utils'
+import { SMALL_USER_CARD } from '../utils/SynapseConstants'
+import { useSynapseContext } from '../utils/SynapseContext'
+import { UserGroupHeader } from '../utils/synapseTypes'
+import TeamBadge from './TeamBadge'
+import UserCard from './UserCard'
+
+type UserOrTeamBadgeProps = {
+  /* The principal ID of the user or team. Required if userGroupHeader is undefined. */
+  principalId?: string | number
+  /* The userGroupHeader of the user or team. Required if principalId is undefined. */
+  userGroupHeader?: UserGroupHeader
+  showFullName?: boolean
+  disableHref?: boolean
+}
+
+export default function UserOrTeamBadge(props: UserOrTeamBadgeProps) {
```

Comment:
Extracted from UserSearchBox and with some additional behavior for fetching/loading